uwblueprint/focus-on-nature
# Review Registration Personal Info

## Notion ticket link

[Registrant Experience: Review Information Step](https://www.notion.so/uwblueprintexecs/Registrant-Experience-Review-Information-Step-cce02b4224c84adca06300508c1dabf6)



## Implementation description
* **Note**: This is the first half of the ticket- reviewing personal info (step 1), I'll make another PR for reviewing additional info once the additional info step is entirely completed (step 2).
* For the editing mechanism, I have a box that changes it's z-index every time you click edit (moves to the background when you're editing, and moves to the foreground after you've saved/cancelled).
* I created to new components- `EditCamperCard` and `EditContactCard` which add onto the `CamperCard` and `ContactCard` components that already exist, so there are some similarities between them (but unfortunately couldn't find a way to use a single, abstracted parent component for both, mentioned below too).
* I use the same reducers that the `CamperCard` and `ContactCard`s use.



## Steps to test
1. Try to register for any camp, i.e. http://localhost:3000/register/camp/63139c7bc3d7b55b44a01531.
2. Fill out the first three steps and navigate to step four.
3. Test out editing, cancelling, and clicking next on this review page, ensuring that the changes you make actually persist (you can verify this by saving a change and going back to step one and seeing the changes).


https://user-images.githubusercontent.com/30396273/209782347-1c7dd31d-91e9-4401-8f59-f1c6751f7ee7.mp4




## What should reviewers focus on?
* I tried to abstract some components to make them more general, but couldn't do so in some places (listed below):
* I couldn't use the accordion item we built in the `common` folder since that's meant specifically for questions, but since the button design is basically the same, I abstracted that part away and made a `GeneralAccordionButton` component, thoughts?
* Although I heavily use the ContactCard and CamperCard components in `.../RegistrantExperience/RegistrationSteps/PersonalInfo`, it's hard to convert this into a single parent component that can also be edited, since my use case for the review card adds on quite a bit of functionality (hence I made `EditContactCard and EditCamperCard components`)
* Upon editing a couple things and clicking cancel, I made a delete modal appear, but this isn't in the designs- I just thought we should probably have it since we do this in a lot of places, thoughts?
* The final review page technically has two sub pages- the review part is what this ticket implements, and after clicking next on the review part, the registrant is brought to the view cart part (which is still the final review page, as indicated by the nav stepper).
* Ensure everything else works properly, with changes persisting when needed.


## Checklist
